<launch>
  <arg name="node_name" default="match_template" />
  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />
  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="false" doc="Specify whether the node displays a window to show edge image" />
  <arg name="match_method" default="5" doc="Matching method. 0:CV_TM_SQDIFF 1:CV_TM_SQDIFF_NORMED 2:CV_TM_CCORR 3:CV_TM_CCORR_NORMED 4:CV_TM_CCOEFF 5:CV_TM_CCOEFF_NORMED" />
  <arg name="template_file" default="" doc="filename for template image"/>
  <arg name="mask_file" default="" doc="filename for mask image"/>

  <!-- match_template_nodelet.cpp -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="match_template" output="screen">
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="match_method" value="$(arg match_method)" />
    <param name="use_mask" value="false" />
    <param name="template_file" value="$(arg template_file)" />
    <param name="mask_file" value="$(arg mask_file)" />
    <param name="match_threshold" value="0.0"/>
  </node>
</launch>
